op-conductor/rpc/ws/server.go

```diff
@@ -0,0 +1,231 @@
+package ws
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/coder/websocket"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// Hub maintains the set of active clients and broadcasts messages to them
+type Hub struct {
+	// Registered clients
+	clients map[*Client]bool
+
+	// Register requests from the clients
+	register chan *Client
+
+	// Unregister requests from clients
+	unregister chan *Client
+
+	// Inbound messages to broadcast to the clients
+	broadcast chan []byte
+
+	// Signal to stop the hub
+	done chan struct{}
+
+	// Logger
+	log log.Logger
+}
+
+// newHub creates a new hub
+func newHub() *Hub {
+	return &Hub{
+		broadcast:  make(chan []byte, 256),
+		register:   make(chan *Client),
+		unregister: make(chan *Client),
+		clients:    make(map[*Client]bool),
+		done:       make(chan struct{}),
+		log:        log.New("component", "websocket-hub"),
+	}
+}
+
+// run starts the hub's main loop
+func (h *Hub) run() {
+	for {
+		select {
+		case <-h.done:
+			// Close all remaining client connections
+			for client := range h.clients {
+				client.Close()
+				delete(h.clients, client)
+			}
+			return
+		case client := <-h.register:
+			h.clients[client] = true
+			clientCount := len(h.clients)
+			h.log.Info("Client registered with hub", "totalClients", clientCount)
+		case client := <-h.unregister:
+			if _, ok := h.clients[client]; ok {
+				delete(h.clients, client)
+				client.Close()
+			}
```

Comment:
Would nice to record registering/deregistering clients as a part of a gauge.

Relates to https://github.com/ethereum-optimism/optimism/pull/15681#discussion_r2128173753